### PR TITLE
Fix bug in dashboard

### PR DESCRIPTION
**Pull request by carlobeltrame**
When the logged in user has an activity responsibility in an activity, and an invited user is also responsible in the same activity, this would lead to errors like "user is undefined", "user is not a function" etc. depending on the browser.
Fixes https://sentry.io/organizations/ecamp/issues/3519676536/

To view an example live on dev.ecamp3.ch, log in using MiData / hussein_kohlmann@hitobito.example.com / hito42bito and visit the dashboard of the past PfiLa 22 Test.

### CODE REVIEW

**Review by BacLuc — APPROVED** [review]

File: frontend/src/components/print/print-react/components/picasso/Responsibles.jsx [hunk 1]
```diff
@@ -5,15 +5,24 @@ import { View, Text } from '../../reactPdf.js'
 const fontSize = 8
 const initialFontSize = fontSize * 0.75
 
+function displayNameFor(campCollaboration) {
```
Comment (usu) [inline frontend/src/components/print/print-react/components/picasso/Responsibles.jsx]: We already have `campCollaborationDisplayName` in `/common/helpers` and `responsiblesCommaSeparated` in `/print/helpers`. Could we use any of these?

File: frontend/src/views/camp/Dashboard.vue [hunk 2]
```diff
@@ -180,14 +180,21 @@ export default {
...
       return activityResponsibles.some(
-        (ar) => ar.campCollaboration().user().id === authUser.id
+        (ar) => this.userIdOrInviteEmail(ar) === authUser.id
       )
     },
-    sortCampCollaborations(campCollaborations) {
-      return campCollaborations.sort(
-        (a, b) => parseInt(a.user().id, 16) - parseInt(b.user().id, 16)
+    sortActivityResponsibles(activityResponsibles) {
+      return activityResponsibles.sort(
```
Comment (usu) [inline frontend/src/views/camp/Dashboard.vue]: I was stumbling over this recently when trying to debug the reactivity issues with vue2.7:
`Array.sort()` performs the sorting algorithm in place, so we're actually directly mutating items in the vuex store. This is not very nice.

Better to use `lodash.sortBy()` or copy the array before sorting `[...activityResponsibles].sort()`

**Review by usu — CHANGES_REQUESTED** [review]

**Comment by carlobeltrame** [comment]
Thanks for the good points. This kind of opened a rabbit hole... There was lots of duplicated code in the print service as well. So I went in and moved it all to common. But then I could not just leave this logic there untested, so I had to set up unit tests for common helpers. And then I noticed some bugs with the inactive indicator and with emoji in user displayNames, so I had to fix the previous implementation in common and add a new dependency. And that's how we went from 2 files changed and a simple bug fix to 31 files changed and a medium-sized refactoring of old tech debt :sweat_smile: 

At first I also wanted to add a regression test for the bug in the Dashboard component, but that turned out to be way too much setup work, and refactoring the Dashboard component into smaller components and writing tests for these is a topic for another PR.

File: frontend/src/components/print/print-react/components/picasso/Responsibles.jsx (hunk 1 above)
Comment (carlobeltrame) [reply frontend/src/components/print/print-react/components/picasso/Responsibles.jsx]: Fixed in e031d55

File: frontend/src/views/camp/Dashboard.vue (hunk 2 above)
Comment (carlobeltrame) [reply frontend/src/views/camp/Dashboard.vue]: Fixed in 20a8ef9

**Review by usu — APPROVED** [review]
Nice 👍 
Thanks for going down the 🐰 -hole... :

**Pull request merged**